pkg/hubble/metrics/api/api.go

```diff
@@ -146,3 +128,101 @@ var registry = NewRegistry(
 func DefaultRegistry() *Registry {
 	return registry
 }
+
+type ContextValues []string
+
+// ContextOptions is a structure used for configuring Hubble metrics context options
+type ContextOptionConfig struct {
+	Name   string        `json:"name,omitempty" yaml:"name,omitempty"`
+	Values ContextValues `json:"values,omitempty" yaml:"values,omitempty"`
+}
+
+// MetricConfig represents a Hubble metric, its options and which resources it applies to.
+// It can hold data parsed from the "hubble-metrics-config" K8S ConfigMap.
+type MetricConfig struct {
+	// Name of the metric
+	Name                 string                 `json:"name,omitempty" yaml:"name,omitempty"`
+	ContextOptionConfigs []*ContextOptionConfig `json:"contextOptions,omitempty" yaml:"contextOptions,omitempty"`
+	// IncludeFilters controls which resources the metric applies to
+	IncludeFilters []*pb.FlowFilter `json:"includeFilters,omitempty" yaml:"includeFilters,omitempty"`
+	// IncludeFilters controls which resources the metric doesn't apply to
+	ExcludeFilters []*pb.FlowFilter `json:"excludeFilters,omitempty" yaml:"excludeFilters,omitempty"`
+}
+
+// DynamicMetricsConfig represents the structure used for configuring
+// Hubble metrics context options
+type DynamicMetricsConfig struct {
```

Comment:
Nit: the name here feels a bit off. Whether or not it's dynamic it somewhat orthogonal to the definition itself. Maybe just `Config`?

Reply:
Sounds good, renamed to `Config`

Reply:
This seems to fail in two ways:
- the linter rejects it with: `api.Config is missing field Metrics` - https://github.com/cilium/cilium/actions/runs/10893022561/job/30227106551?pr=34849
- bpf code fails to build 
`bash: line 1: ccache: command not found make[1]: *** [Makefile:255: bpf_lxc.o] Error 127`

Went with `MetricsConfig`: plural 's' mid-word, for this wrapper type containing a slice of `MetricConfig`s

Reply:
`exhaustruct` is relatively new, and you probably can indicate the fields are optional to handle that.

The bpf code error is certainly an unrelated issue.

Reply:
Added ignore annotations and rebased on main - all resolved.